.github/workflows/translations.yaml

```diff
@@ -10,13 +10,27 @@ on:
       - dev
     paths:
       - "**strings.json"
+      - ".github/workflows/translations.yaml"
 
 env:
   DEFAULT_PYTHON: 3.9
 
 jobs:
+  secrets:
+    name: Check for secrets
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check for secrets.LOKALISE_TOKEN
+        shell: bash
+        run: |
+          if [ -n "${{ secrets.LOKALISE_TOKEN }}" ]; then
+            echo "::set-output name=lokalise::1"
+          fi
```

Comment:
It is not very common to check for secrets like this. We don't do it anywhere else either. Why add it here?

Reply:
Because otherwise my fork gets lots of annoying errors when workflows run.

This really is the correct way to guard things. The alternative is to guard by repository organization, but then if someone wants to test in their form, they can't simply add secrets, they also have to edit workflows.

Reply:
Maybe this should not run on forks at all?

Reply:
Up to you. I don't know enough about the project to have a strong opinion.

Personally when actively working on things, I like being able to test in my forks. This approach allows that.

Reply:
> I don't know enough about the project to have a strong opinion.

I suggest you work on smaller things first until you know enough about the project to be able to improve our CI

Reply:
I'm certain that this change would improve your CI.

Whether you ever expect non members to work on CI changes is something I don't know.

As I've stated, as a non member, it's helpful to be able to test changes to CI before proposing them, and as such, it's helpful to be able to provide my own secrets and have code run.

That said, in general, when things will not work in forks, they shouldn't run in forks, so as to avoid a run of :x:s and their corresponding notifications.

Reply:
What if we guarded the repository organization but only when the job was run on a schedule? So something like:
```yaml
if: github.repository.owner == 'home-assistant' || !github.event.schedule
```
This solves the flood of annoying errors problem since that primarily comes from the jobs that run daily and fail to find secrets. And it allows users which actually want to run those workflows in their forks after they have filled in the secrets to do so manually via `workflow_dispatch` (or by creating a PR in their fork).

The problem with your solution is it actually makes it quite difficult for a developer that really wanted to run those workflows. The workflow will appear to succeed when a secret is missing when really it didn't do anything. Any developer that actually wanted to run that workflow for real would much rather see an error when a secret is missing. Better to prevent unintentional runs (the scheduled ones) and let intentional runs proceed as normal IMO.